### ISSUE TITLE
Enable the service raw id / functional id matching on API endpoints

### ISSUE THREAD

**Issue by lbroudoux** [microcks/microcks#1225]
### Reason/Context

This comes from https://github.com/microcks/microcks-testcontainers-java/issues/51

`serviceId` as used in the API path has —most of the time— two different meanings. It can be the raw unique identifier used in the database to identify the service, OR it can be the `service_name:service_version` identifier.

In the case of this API (the operation properties update), the translation from `service_name:service_version` to raw identifier has not been put in place. So basically, Microcks fails to retrieve a service with this ID and then apply the update to the operation.

### Description

So basically, we would have to generalize this translation everywhere - not just on the `/api/tests` or `/api/services/{serviceId}` endpoints - but also on all the derivatives.

**Comment by frademacher** [comment]
Thanks for the investigation and elaboration here and over at the Microcks testcontainers issue tracker, @lbroudoux!

For me, the workaround mentioned in https://github.com/microcks/microcks-testcontainers-java/issues/51 (fetch and use service database ID) is perfectly fine for now and I'll test it ASAP.

As for extending Microcks OpenAPI importer with recognizing `x-microcks` extensions related to constraints, this would be the ideal solution for my use case (at least from my perspective). Having "everything" in one spec has IMHO the charm that one can use exactly the spec as their single source of truth w.r.t. mock preparation and provisioning. But as said, for me the mentioned workaround is also perfectly fine for now :-)

**Comment by frademacher** [comment]
I finally came to check whether the workaround mentioned in https://github.com/microcks/microcks-testcontainers-java/issues/51 (fetch and use service database ID) works for me and can confirm that this is the case.

So for anyone having issues with specifying parameter constraints for operations using Microcks' API, the following requests work (in the context of mocking the example OpenAPI spec from https://github.com/microcks/microcks-testcontainers-java/issues/51):

```bash
MICROCKS_PORT=32905

# Fetch Micrcocks service ID
SERVICE_ID=$(curl -s --location "http://localhost:$MICROCKS_PORT/api/services/encoding_test_api:1" --header 'Accept: application/json' | jq -r ".service.id")

# Invoke Micrcocks' Override Service Operation endpoint with the fetched service ID to specify a header constraint for the example API's GET /common/genders operation
curl --location --request PUT "http://localhost:$MICROCKS_PORT/api/services/$SERVICE_ID/operation?operationName=GET%20%2Fcommon%2Fgenders" \
--header 'Content-Type: application/json' \
--data '{  
  "parameterConstraints": [
    {
      "name": "mobileTokenHeader",
      "required": "true",
      "recopy": "false",
      "mustMatchRegexp": "someToken",
      "in": "header"
    }    
  ]
}'
```

**Comment by lbroudoux** [comment]
`/api/services/{id}/ endpoints tested and validated in the context of microcks/microcks-operator#52

**Comment by lbroudoux** [comment]
Now implemented.
